api-catalog-services/src/main/java/com/ca/mfaas/apicatalog/swagger/api/ApiDocV2Service.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package com.ca.mfaas.apicatalog.swagger.api;
+
+import com.ca.mfaas.apicatalog.services.cached.model.ApiDocInfo;
+import com.ca.mfaas.apicatalog.swagger.ApiDocTransformationException;
+import com.ca.mfaas.product.gateway.GatewayClient;
+import com.ca.mfaas.product.gateway.GatewayConfigProperties;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.swagger.models.ExternalDocs;
+import io.swagger.models.Path;
+import io.swagger.models.Scheme;
+import io.swagger.models.Swagger;
+import io.swagger.util.Json;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.validation.UnexpectedTypeException;
+import java.io.IOException;
+import java.util.*;
+
+@Slf4j
+public class ApiDocV2Service extends AbstractApiDocService<Swagger, Path> {
+
+    public ApiDocV2Service(GatewayClient gatewayClient) {
+        super(gatewayClient);
+    }
+
+    public String transformApiDoc(String serviceId, ApiDocInfo apiDocInfo) {
+        Swagger swagger;
+
+        try {
+            swagger = Json.mapper().readValue(apiDocInfo.getApiDocContent(), Swagger.class);
+        } catch (IOException e) {
```

Comment:
This exception will not be thrown. If IOException is to be thrown from apiDocInfo.getApiDocContent() it will be caught earlier in class TransformApiDocService(line 50)